private/tools/java/com/github/bazelbuild/rules_jvm_external/maven/MavenPublisher.java

```diff
@@ -174,10 +203,27 @@ private static CompletableFuture<Void> upload(
     uploads.add(upload(String.format("%s%s.md5", base, append), credentials, md5));
     uploads.add(upload(String.format("%s%s.sha1", base, append), credentials, sha1));
 
-    if (gpgSign) {
-      uploads.add(upload(String.format("%s%s.asc", base, append), credentials, sign(item)));
-      uploads.add(upload(String.format("%s%s.md5.asc", base, append), credentials, sign(md5)));
-      uploads.add(upload(String.format("%s%s.sha1.asc", base, append), credentials, sign(sha1)));
+    SigningMethod signingMethod = signingMetadata.getSigningMethod();
+    if (signingMethod.equals(SigningMethod.GPG)) {
```

Comment:
We've lost the code path where there is no signing to be done

Reply:
The current implementation is 
```
if (signingMethod.equals(SigningMethod.GPG)) {
    // do somthing
} else if (signingMethod.equals(SigningMethod.PGP)) {
    // do something
}

return CompletableFuture.allOf(uploads.toArray(new CompletableFuture<?>[0]));
```

where signing method is defined in 
```
      if (gpgSign) {
        this.signingMethod = SigningMethod.GPG;
      } else if (useInMemoryPgpKeys) {
        this.signingMethod = SigningMethod.PGP;
      } else {
        this.signingMethod = SigningMethod.NONE;
      }
```

Therefore, we still keeps the no-signing path as before where `signingMethod = SigningMethod.NONE`